examples/lib/stories/collision_detection/multiple_shapes.dart

```diff
@@ -0,0 +1,242 @@
+      : super(position, size, velocity) {
+    final shape = HitboxPolygon([
+      Vector2(-1.0, 0.0),
+      Vector2(-0.8, 0.6),
+      Vector2(0.0, 1.0),
+      Vector2(0.6, 0.9),
+      Vector2(1.0, 0.0),
+      Vector2(0.6, -0.8),
+      Vector2(0, -1.0),
+      Vector2(-0.8, -0.8),
+    ]);
+    addShape(shape);
+  }
+}
+
+class CollidableRectangle extends MyCollidable {
+  CollidableRectangle(Vector2 position, Vector2 size, Vector2 velocity)
+      : super(position, size, velocity) {
+    addShape(HitboxRectangle());
+  }
+}
+
+class CollidableCircle extends MyCollidable {
+  CollidableCircle(Vector2 position, Vector2 size, Vector2 velocity)
+      : super(position, size, velocity) {
+    final shape = HitboxCircle();
+    addShape(shape);
+  }
+}
+
+class SnowmanPart extends HitboxCircle {
+  static const startColor = Colors.white;
+  final hitPaint = Paint()
+    ..color = startColor
+    ..strokeWidth = 1
+    ..style = PaintingStyle.stroke;
+
+  SnowmanPart(double definition, Vector2 relativePosition, Color hitColor)
+      : super(definition: definition) {
+    this.relativePosition.setFrom(relativePosition);
+    onCollision = (Set<Vector2> intersectionPoints, HitboxShape other) {
+      if (other.component is ScreenCollidable) {
+        hitPaint..color = startColor;
+      } else {
+        hitPaint..color = hitColor;
+      }
+    };
+  }
+
+  @override
+  void render(Canvas canvas, Paint paint) {
+    super.render(canvas, hitPaint);
+  }
+}
+
+class CollidableSnowman extends MyCollidable {
+  CollidableSnowman(Vector2 position, Vector2 size, Vector2 velocity)
+      : super(position, size, velocity) {
+    rotationSpeed = 0.2;
+    final top = SnowmanPart(0.4, Vector2(0, -0.8), Colors.red);
+    final middle = SnowmanPart(0.6, Vector2(0, -0.3), Colors.yellow);
+    final bottom = SnowmanPart(1.0, Vector2(0, 0.5), Colors.green);
+    addShape(top);
+    addShape(middle);
+    addShape(bottom);
+  }
+}
+
+class MultipleShapes extends BaseGame
+    with HasCollidables, HasDraggableComponents {
+  final TextConfig fpsTextConfig = TextConfig(
+    color: BasicPalette.white.color,
+  );
+
+  @override
+  Future<void> onLoad() async {
+    final screen = ScreenCollidable();
+    final snowman = CollidableSnowman(
+      Vector2.all(150),
+      Vector2(100, 200),
+      Vector2(-100, 100),
+    );
+    MyCollidable lastToAdd = snowman;
+    add(screen);
+    add(snowman);
+    var totalAdded = 1;
+    while (totalAdded < 20) {
+      lastToAdd = createRandomCollidable(lastToAdd);
+      final lastBottomRight =
+          lastToAdd.toAbsoluteRect().bottomRight.toVector2();
+      if (screen.containsPoint(lastBottomRight)) {
+        add(lastToAdd);
+        totalAdded++;
+      } else {
+        break;
+      }
+    }
+  }
+
+  final _rng = Random();
+  final _distance = Vector2(100, 0);
+
+  MyCollidable createRandomCollidable(MyCollidable lastCollidable) {
+    MyCollidable collidable;
```

Comment:
nit: I would declare this right above the switch, where it's used

Reply:
Ah, it is not even needed.